driven/storage/adapter_scores.go

```diff
@@ -20,18 +20,30 @@ import (
 	"github.com/rokwire/rokwire-building-block-sdk-go/utils/errors"
 	"github.com/rokwire/rokwire-building-block-sdk-go/utils/logging/logutils"
 	"go.mongodb.org/mongo-driver/bson"
-	"go.mongodb.org/mongo-driver/mongo/options"
+	"go.mongodb.org/mongo-driver/mongo"
 )
 
 // GetScore finds score object for user
 func (a *Adapter) GetScore(orgID string, appID string, userID string) (*model.Score, error) {
-	filter := bson.M{"org_id": orgID, "app_id": appID, "user_id": userID}
-	var entry model.Score
-	err := a.db.scores.FindOne(a.context, filter, &entry, nil)
-	if err != nil {
-		return nil, errors.WrapErrorAction(logutils.ActionFind, model.TypeScore, filterArgs(filter), err)
+	pipeline := mongo.Pipeline{
+		bson.D{{Key: "$match", Value: bson.M{"org_id": orgID, "app_id": appID}}},
+		// add rank
+		bson.D{{Key: "$setWindowFields", Value: bson.M{
+			"sortBy": bson.M{"score": -1},
+			"output": bson.M{
+				"rank": bson.M{"$rank": bson.M{}},
```

Comment:
I think we might want to use [$denseRank](https://www.mongodb.com/docs/manual/reference/operator/aggregation/denseRank/#mongodb-group-grp.-denseRank) instead of `$rank` here to avoid results like the following where ranks are skipped:

![image](https://github.com/user-attachments/assets/c7859fcf-dab3-455f-8f1f-00b6ab13c67f)